src/_pytest/python.py

```diff
@@ -1343,6 +1344,8 @@ def _idval(
         # Name of a class, function, module, etc.
         name: str = getattr(val, "__name__")
         return name
+    elif isinstance(val, pathlib.Path):
+        return str(val)
```

Comment:
we might want to consider the os.fspath protocol here ?

Reply:
> we might want to consider the os.fspath protocol here ?

I'm not wholly familiar with this, would you mind expanding on what you mean?

After a brief browse of  the docs, maybe making this more general with something like this?

```python
elif isinstance(val, os.PathLike):
    return os.fspath(val)
 ```

Reply:
Yep, if it works

Reply:
`mypy` didn't like my attempt (though everything else passed):

```python
    elif isinstance(val, os.PathLike):
        path = os.fspath(val)
        if isinstance(path, bytes):
            return path.decode("UTF-8")
        else:
            return path
```

`mypy` output:

```
src/_pytest/python.py:1346: error: Only @runtime_checkable protocols can be used with instance and class checks  [misc]
src/_pytest/python.py:1351: error: Returning Any from function declared to return "str"  [no-any-return]
```

I believe the first error is a bug that has been fixed in `typeshed` but released in a version that hasn't yet been bundled into `mypy` https://github.com/python/typeshed/issues/4681. I'm not sure about the second since [from the docs](https://docs.python.org/3/library/os.html#os.fspath) it looks like `fspath` should return one of `str` or `bytes` (see also the [typeshed definition](https://github.com/python/typeshed/blob/master/stdlib/3/os/__init__.pyi#L336))

Reply:
Had the same situation today: https://github.com/pytest-dev/pytest/pull/8174/files#diff-df52f8f6a3544754cc8ebdf903594738e68a18dc9ac3c959f646cf4705a9afedR303-R305 I suggest doing the same, adding a type ignore until the next mypy release.

As for the bytes, you can probably treat it like `bytes` itself is treated -- see `if isinstance(val, STRING_TYPES)` branch above.

Reply:
Thanks @bluetech! That did the trick (I just copy-pasted your TODO comment for consistency/easier greping), the mixed handling for `STRING_TYPES` you suggested also allowed me to just remove the branching, see 
a6c3c5d

Reply:
Now that the handling is more general here, it might be worth adding more test cases (outside of just `pathlib.Path`), though I'm not very informed/creative in this regard, so I'd be happy to hear suggestions.